extensions/wikia/DiscussionModeration/ReportedPostsHelper.class.php

```diff
@@ -90,4 +90,108 @@ private function buildPermalink( Uri $uri ) {
 
 		return $this->baseDomain . $this->scriptPath . '/wikia.php?' . build_query( $controllerQueryParams );
 	}
+
+	public function applyBadges( array &$body ) {
+		$posts = $body['_embedded']['doc:posts'] ?? [];
+		$contributors = $body['_embedded']['contributors'][0]['userInfo'] ?? [];
+		$userIds = [];
+
+		foreach ( $posts as $post ) {
+			if ( isset( $post['createdBy']['id'] ) ) {
+				$userIds[] = (int)$post['createdBy']['id'];
+			}
+			if ( isset( $post['lastEditedBy']['id'] ) ) {
+				$userIds[] = (int)$post['lastEditedBy']['id'];
+			}
+			if ( isset( $post['lastDeletedBy']['id'] ) ) {
+				$userIds[] = (int)$post['lastDeletedBy']['id'];
+			}
+			if ( isset( $post['threadCreatedBy']['id'] ) ) {
+				$userIds[] = (int)$post['threadCreatedBy']['id'];
+			}
+
+			if ( isset( $post['_embedded']['thread'][0]['firstPost']['createdBy']['id'] ) ) {
+				$userIds[] = (int)$post['_embedded']['thread'][0]['firstPost']['createdBy']['id'];
+			}
```

Comment:
Does first post have `lastEditedBy` and `lastDeletedBy` fields as well?

Reply:
nope

Reply:
at least I did not see it

Reply:
but according to discussion's code it can, good catch!